File: src/main/java/com/realyoungk/sdi/repository/VisitRepository.java
package com.realyoungk.sdi.repository;


import com.realyoungk.sdi.dto.VisitDto;
import org.springframework.stereotype.Repository;

import java.util.Date;
import java.util.List;

@Repository
public interface VisitRepository {
    List<VisitDto> findByStartedAtAfterOrderByStartedAtDesc(Date startedAt);

    VisitDto save(VisitDto visitDto);
}

File: src/main/kotlin/com/realyoungk/sdi/repository/VisitRepository.kt
package com.realyoungk.sdi.repository

import com.realyoungk.sdi.entity.VisitEntity
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface VisitRepository {
    fun findByStartedAtAfterOrderByStartedAtDesc(startedAt: Date): List<VisitEntity>
    
    fun save(visitEntity: VisitEntity): VisitEntity
}